File: app/src/main/java/app/we/go/emojidraw/features/practice/PracticeContract.java
package app.we.go.emojidraw.features.practice;

import java.util.List;

import app.we.go.emojidraw.model.Stroke;
import app.we.go.emojidraw.arch.mvp.MVP;

public interface PracticeContract {

    interface View extends MVP.View {

        void onGuessesReturned(List<String> strings);

        void setEmojiToDraw(String emoji, String emojiDescription);

        void onEmojiDrawnCorrectly(String emoji);

        void onAllEmojisDrawn();

        void onAllEmojisDrawnWithCheat();

        void onTimeOut();

        void showTooltip(int position);

        void hideTooltip();

        void showErrorMessage();
    }

    interface Presenter extends MVP.Presenter<View> {

        void onStrokeAdded(List<Stroke> strokes);

        void onTimeExpired();

        void onSkip();

    }

}

File: app/src/main/java/app/we/go/emojidraw/features/practice/PracticeContract.kt
package app.we.go.emojidraw.features.practice

import app.we.go.emojidraw.arch.mvp.MVP
import app.we.go.emojidraw.model.Stroke

interface PracticeContract {

    interface View : MVP.View {

        fun onGuessesReturned(strings: List<String>)

        fun setEmojiToDraw(emoji: String, emojiDescription: String)

        fun onEmojiDrawnCorrectly(emoji: String)

        fun onAllEmojisDrawn()

        fun onAllEmojisDrawnWithCheat()

        fun onTimeOut()

        fun showTooltip(position: Int)

        fun hideTooltip()

        fun showErrorMessage()
    }

    interface Presenter : MVP.Presenter<View> {

        fun onStrokeAdded(strokes: List<Stroke>)

        fun onTimeExpired()

        fun onSkip()

    }

}
